### PR TITLE
[ML] Update Linux builder Docker container for 9.0

### PR DESCRIPTION
**Pull request by valeriy42**
This PR updates all Linux-related Dockerfiles and tutorials with the latest dependencies, libraries, and C++ compiler versions. 

### Changes

- **Dockerfile Updates**:
  - The base image is changed from `centos:7` to `rockylinux:8` to maintain compatibility with RHEL in the future.
  - Updated all GCC versions to use GCC 13.3 and binutils 2.41 for enhanced C++17 support and security hardening.
  - Other updated libraries:
    - `binutils` 2.37 -> 2.41
    - `libxml2` 2.9.14 -> 2.10.4.
    - `patchelf` 0.13 -> 0.18.0
    - `python` 3.10.0 -> 3.12.7
    - `cmake` 3.19.2 -> 3.30.5

- **Tutorial Updates**:
  - Revised instructions for setting up the Linux build environment to match new dependencies and configurations.
  - Provided updated example commands for package installation compatible with Rocky Linux 8 and similar distributions.
- **Buildkite**:
  - Updated the CI scripts to use the newest version of the Docker containers.

### CODE REVIEW

File: build-setup/linux.md [hunk 1]
```diff
@@ -8,26 +8,30 @@ You will need the following environment variables to be defined:
...
+export PATH=$JAVA_HOME/bin:/usr/local/gcc133/bin:/usr/local/cmake/bin:/usr/bin:/bin:/usr/sbin:/sbin:/home/vagrant/bin
 # Only required if building the C++ code directly using cmake - adjust depending on the location of your Git clone
 export CPP_SRC_HOME=$HOME/ml-cpp
 ```
 
 ### OS Packages
 
-You need the C++ compiler and the headers for the `zlib` library that comes with the OS.  You also need the archive utilities `unzip`, `bzip2` and `xz`. `libffi-devel` and `openssl-devel` are dependencies for building PyTorch. Finally, the unit tests for date/time parsing require the `tzdata` package that contains the Linux timezone database.  On RHEL/CentOS these can be installed using:
+You need the C++ compiler and the headers for the `zlib` library that comes with the OS. You also need the archive utilities `unzip`, `bzip2` and `xz`. `libffi-devel` and `openssl-devel` are dependencies for building PyTorch. Installing the group `Debelopment Tools` will ensure you habe all required tools to buils the C++ compiler. Finally, the unit tests for date/time parsing require the `tzdata` package that contains the Linux timezone database. On RHEL/Rocky Linux these can be installed using:
```
Comment (edsavage) [inline build-setup/linux.md:27]: nit: Debelopment -> Development
nit: habe -> have
nit: buils -> build

File: build-setup/linux.md [hunk 2]
```diff
@@ -239,58 +243,58 @@ to complete the build.
 
 ### CMake
 
-CMake version 3.19.2 is the minimum required to build ml-cpp. Download version 3.23.2 from <https://github.com/Kitware/CMake/releases/download/v3.23.2/cmake-3.23.2-Linux-x86_64.sh> and install:
+CMake version 3.30.5 is the minimum required to build ml-cpp. Download version 3.30.5 from <https://github.com/Kitware/CMake/releases/download/v3.30.5/cmake-3.30.5-linux-x86_64.sh> and install:
```
Comment (edsavage) [inline build-setup/linux.md:246]: Is CMake 3.30.5 the actual minimum version? Or just the one used here?

File: dev-tools/docker/linux_dependency_builder_image/Dockerfile [hunk 3]
```diff
@@ -10,64 +10,65 @@
 #
 
 # Increment the version here when a new tools/3rd party components image is built
-FROM docker.elastic.co/ml-dev/ml-linux-build:30 AS builder
+FROM docker.elastic.co/ml-dev/ml-linux-build:31 AS builder
 
 # This is basically automating the setup instructions in build-setup/linux.md
 
-MAINTAINER Ed Savage <ed.savage@elastic.co>
+LABEL maintainer="Ed Savage <ed.savage@elastic.co>"
```
Comment (edsavage) [inline dev-tools/docker/linux_dependency_builder_image/Dockerfile:17]: For consistency, maybe change the maintainer to be you here as well?

File: dev-tools/docker/linux_image/Dockerfile [hunk 4]
```diff
@@ -9,134 +9,157 @@
...
-  yum repolist enabled && \
-  rm /var/lib/rpm/__db.* && \
-  yum install -y bzip2 gcc gcc-c++ git libffi-devel make texinfo unzip wget which xz zip zlib-devel
+RUN dnf -y update && \
+    dnf install -y dnf-plugins-core && \
+    dnf config-manager --set-enabled powertools && \
+    dnf groupinstall "Development Tools" -y && \
+    dnf install -y bzip2 gcc gcc-c++ git libffi-devel make texinfo unzip wget which xz zip zlib-devel && \
+    dnf clean all && \
+    rm -rf /var/cache/dnf /tmp/*
```
Comment (edsavage) [inline dev-tools/docker/linux_image/Dockerfile:26]: Are these commented out commands still needed here?

**Review by edsavage — APPROVED** [review]
LGTM

I've just left a few comments here and there.

File: build-setup/linux.md (hunk 2 above)
Comment (valeriy42) [reply build-setup/linux.md:246]: It's not a minimum required indeed. It's just a decently recent version to support C++20 features.

**Pull request merged**